src/main/java/org/openpnp/machine/reference/vision/ReferenceFiducialLocator.java

```diff
@@ -177,40 +183,49 @@ public Location getHomeFiducialLocation(Location location, Part part) throws Exc
      * @return
      * @throws Exception
      */
-    private static Location getFiducialLocation(BoardLocation boardLocation, Placement fid)
-            throws Exception {
+	private static Location getFiducialLocation(BoardLocation boardLocation, Placement fid, Length fidDia) throws Exception {
         Camera camera = Configuration.get().getMachine().getDefaultHead().getDefaultCamera();
 
         Logger.debug("Locating {}", fid.getId());
 
+		Footprint footprint;
+
+		// Careful...magic number below. Also used in JobPanel. This are for
+		// global panel fiducials, which do not have an entry in the PCB BOM.
+		if ((fid.getId().equals("PanelFid1_a322") || fid.getId().equals("PanelFid2_g301"))) {
```

Comment:
Philosophically, you probably need to answer the question on how the user should enter panel fiducials. To me, __PCB__ fids as part of the BOM make 100% sense, and thus entering them as parts and having corresponding  part which then has a corresponding package makes sense.  In a panel, they show up in every board, just as you'd expect (because the fids are part of the BOM).

__Panel__ fiducials, however, are treated differently in every layout package I've worked on. As you see the screen capture from the Altium package that was posted, there a separate GUI to specify the panel. And that's true in Mentor, diptrace, etc. You have a dedicated dlg to enter all the panel info. 

Now, I opted to add a dlg to handle the panel parameters and part of that was specifying the panel fids locations (pcb fids stay the same). 

if you decide you don't like the panel fids being specified in a separate dlg, then it seems (not certain, however) that you need to deal with specifying panel fids in just one of the boards in the array. That would preserve the way all the software in openpnp works today, but you still need the dlg, and then you have to explain to users that panel fids need to be in one of the panels....in other words, the rules get complicated.

And you already need a dlg to specify the panel parameters....

Re: special IDs....the problem I ran into was that we have the global panel fids the user entered into the panel specification. How to get those all the way down to the RefFidLoc() routine. They were passed as placements (but not part of the list of placements) with special names. And when those special names are encountered down in RefFidLoc(), the method knows to create the footprint with the given diameter. 

If you opt to specify panel fids some other way that that avoids this, I could certainly re-work. But think about the user implications of breaking apart where all this info sits. With packages.xml and parts.xml being treated as global databases, it already gets to be a bit messy because fids (and parts) on Project A that was created by engineer X could very well be overwritten the next day by engineer Y working on Project B (and when engineer A returns to the machine, his work is polluted).  Long term, if packages and parts are moved to per-job, then this concern is lessened.  But right now it gets quite messy even for me on the few boards I've been through thus far. I think it much better to have all the panel details held in the board file instead of spread across 3 files (board, package, part), 1 of which is project-based and the other 2 which are global. 

The other items I can certainly tidy up, but first please share your thoughts on how you'd like to see the user enter the location of the panel fids